src/Polly.Core/Utils/Pipeline/ComponentWithDisposeCallbacks.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace Polly.Utils.Pipeline;
+
+internal class ComponentWithDisposeCallbacks : PipelineComponent
+{
+    private readonly List<Action> _callbacks;
+
+    public ComponentWithDisposeCallbacks(PipelineComponent component, List<Action> callbacks)
+    {
+        Component = component;
+        _callbacks = callbacks.ToList();
```

Comment:
`ToList()` on a `List<T>` looks a bit weird to me 😄

```suggestion
        _callbacks = new(callbacks);
```

Reply:
changed the code slightly so the copy happens outside of the class